src/core/ActivityInfoTree.cpp

```diff
@@ -283,5 +285,71 @@ void ActivityInfoTree::init()
 
     qmlRegisterSingletonType<QObject>("GCompris", 1, 0, "ActivityInfoTree", menuTreeProvider);
     qmlRegisterType<ActivityInfo>("GCompris", 1, 0, "ActivityInfo");
+
+
+}
+void ActivityInfoTree::filterBySearch(const QString& text)
+{
+
+    if(text.trimmed().length() < m_searchedtext.length() or text.trimmed() == ""){
+        if(text.trimmed() == ""){
+            m_searchedtext = "";
+            m_searched = m_menuTreeFull;
+        }
+
+        m_menuTree = m_menuTreeFull;
+    }
+
+
+
+    if(!text.trimmed().isEmpty() and text.length() != m_searchedtext.length()){
+        m_menuTree.erase(std::remove_if(m_menuTree.begin(),m_menuTree.end(),
+                                        [&](ActivityInfo* activity){
+                             if(!activity->title().contains(text.trimmed(),Qt::CaseInsensitive)){
+                                 if(!activity->name().contains(text.trimmed(),Qt::CaseInsensitive))
+                                       return true;
+                                 else
+                                    return false;
+                             }
+                             else
+                                 return false;
+                         }),m_menuTree.end());
+    }
+
+    else{
+           m_menuTree = m_searched;
+    }
+    m_searchedtext = text;
+    m_searched = m_menuTree;
+    filterEnabledActivities();
+    filterLockedActivities();
+    sortByDifficulty();
+    emit menuTreeChanged();
+
+}
+QVariantList  ActivityInfoTree::allCharacters()
+{
+    int i = 0;
+    QSet<QString> keyboardchars;
+    QString templetter;
+    foreach(auto tree,m_menuTreeFull)
+    {
+        auto title = tree->title();
+        for (int j = 0 ; j!= title.length(); j++)
+        {
+            if(!title.at(j).isSpace() and !title.at(j).isPunct())
+            {
+                templetter = title.at(j).toLower();
```

Comment:
you can directly insert title.at(j).toLower() without having to use a temporary variable